### PR TITLE
Skin upgrade

### PR DESCRIPTION
**Pull request by arpitaagrawal**
Hello Professor  @chrismattmann ,

I have commited the UI changes that i have done(created a new branch for this - skin-upgrade) + pubmed service changes. 
The changes committed till now are not ready to be merged yet, it has a lot of unused code(in js files) which has been replicated from the previous version. I will be removing that as i make progress in integrating the rest of the services.

### CODE REVIEW

**Comment by chrismattmann** [comment]
I'm going to pull this PR into a v2 branch @arpitaagrawal mostly b/c i don't think we should have the logic in the WicketApp java file that has to select v2, etc. It should just work..

**Comment by chrismattmann** [comment]
@arpitaagrawal ping can you please make the updates so we can merge?

**Comment by arpitaagrawal** [comment]
@chrismattmann -Hello Professor,I have made the base class changes required for the version change and have committed the same as well. Please review them.
Also, by mistake i had committed the Wikipedia change in the same PR itself,have reverted that and will be creating a new branch to commit the same- with a new PR Request. 

The change made is as follows:
There is a basehomepage class
Derived HomePageV2 class - for version 2.

Properties in context.xml - org.shangridocs.skin defines version of homepage
shangridocs.homepage.default - defines the default home page

for x = org.shangridocs.skin - there should be a property in context.xml file - shangridocs.homepage.'x' - where x is the skin version . E.g. - x=v2, shangridocs.homepage.v2
if found - then that becomes the homepage else the default page is loaded . i.e - shangridocs.homepage.default

File: shangridocs-webapp/src/main/java/org/shangridocs/BaseHomePage.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package org.shangridocs;
```
Comment (chrismattmann) [inline shangridocs-webapp/src/main/java/org/shangridocs/BaseHomePage.java:18]: ALv2 header

File: shangridocs-webapp/src/main/java/org/shangridocs/HomePageV2.html [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+<title>ShangriDocs</title>
+<!-- Google Fonts -->
+<!-- Bootstrap core CSS -->
+
+<wicket:link>
+	<link rel="stylesheet" href="new_skin_css/css/bootstrap.min.css" />
+	<link rel="stylesheet" href="new_skin_css/style.css" />
+</wicket:link>
+<!-- <link href="bower_components/bootstrap/dist/css/bootstrap.min.css"
```
Comment (chrismattmann) [inline shangridocs-webapp/src/main/java/org/shangridocs/HomePageV2.html]: if we don't need it, remove it.

File: shangridocs-webapp/src/main/java/org/shangridocs/HomePageV2.html [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+	rel="stylesheet" type="text/css" />
+<link href="style.css" rel="stylesheet" type="text/css" /> -->
```
Comment (chrismattmann) [inline shangridocs-webapp/src/main/java/org/shangridocs/HomePageV2.html]: same here, remove

File: shangridocs-webapp/src/main/java/org/shangridocs/HomePageV2.html [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+	<!-- Bootstrap core JavaScript
+================================================== -->
+	<!-- Placed at the end of the document so the pages load faster -->
+
+	<wicket:link>
+		<script type="text/javascript" src="jquery.min.js"></script>
+		<script type="text/javascript" src="bootstrap.min.js"></script>
+		<script type="text/javascript" src="bootstrap-tour.min.js"></script>
+		<script type="text/javascript" src="homePage_wicket_v2.js"></script>
+		<!-- <script type="text/javascript" src="homePage_v2.js"></script> -->
```
Comment (chrismattmann) [inline shangridocs-webapp/src/main/java/org/shangridocs/HomePageV2.html]: remove

File: shangridocs-webapp/src/main/java/org/shangridocs/HomePageV2.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
+package org.shangridocs;
```
Comment (chrismattmann) [inline shangridocs-webapp/src/main/java/org/shangridocs/HomePageV2.java:18]: ALv2 header

File: shangridocs-webapp/src/main/webapp/META-INF/context.xml [hunk 6]
```diff
@@ -17,11 +17,14 @@ the License.
 -->
 <Context path="/shangridocs">
         
-    <Parameter name="shangridocs.homepage"
-        value="org.shangridocs.HomePage"/>  
+    <Parameter name="shangridocs.homepage.default"
+        value="org.shangridocs.HomePage"/>
+          
+    <Parameter name="shangridocs.homepage.v2"
```
Comment (chrismattmann) [inline shangridocs-webapp/src/main/webapp/META-INF/context.xml]: i don't understand these params, it's kind of confusing. Why not just make shangridocs.homepage reference the page class, and set the V2page to be the default. We don't need to introduce shangridocs.homepage.v2.

**Comment by chrismattmann** [comment]
@arpitaagrawal please separate the pubmed resource update into a separate PR. Then, with respect to the skin update, please add ALv2 headers on anything new, and resubmit this PR. I think the property name additions are confusing. We don't need any new properties. We can reference the home page and we will just ref the V2 home page as default.

**Comment by arpitaagrawal** [comment]
@chrismattmann Professor, I have incorporated the changes. I will be pushing the rest of the changes i.e. integration of pubmed, wikipedia service with UI once these changes are merged. As, the integration changes are made on top of these changes and hence cannot be pushed as a separate pull request until this is merged.

**Comment by nunsuch123** [comment]
Please remove nunsuch123 from this repository. 

Lauren

**Comment by chrismattmann** [comment]
will do @nunsuch123 !

**Comment by chrismattmann** [comment]
merged thanks @arpitaagrawal !

**Pull request merged**